src/language-html/utils.js

```diff
@@ -592,14 +592,23 @@ function dedentString(text, minIndent = getMinIndentation(text)) {
         .join("\n");
 }
 
-function countChars(text, char) {
-  let counter = 0;
-  for (let i = 0; i < text.length; i++) {
-    if (text[i] === char) {
-      counter++;
-    }
+function chooseEnclosingQuote(options, value) {
```

Comment:
Is it possible to reuse https://github.com/prettier/prettier/blob/b2fa604e77dab3c9e33e832f639703e7d22f128f/src/common/util.js#L350-L382 ?

Reply:
I propose to modify a bit `getPreferredQuote` in https://github.com/prettier/prettier/pull/11569, so that it can used in the three printers: jsx, html, handlebars.